### PR TITLE
Address adolfos comments

### PR DESCRIPTION
**Pull request by bmagyar**
@graiola

### CODE REVIEW

File: paper.md [hunk 1]
```diff
@@ -72,8 +72,9 @@ bibliography: paper.bib
...
+  In recent years the Robotics Operating System [@quigley2009ros] (ROS) has become the 'de facto' standard framework for the development of software in robotics. The idea of `ros_control` originates from the `pr2_controller_manager` framework specific to the PR2 robot. The primary motivation for a separate framework for robot-control was necessary because ROS doesn't provide a realtime-safe communication layer but the framework implements mechanisms for controller-lifecycle- and hardware resource-  management as well as abstractions on hardware interfaces. The `ros_control` framework provides the capability to implement and manage robot controllers with a focus on both _real-time performance_ and _sharing of controllers_ in a robot-agnostic way.
```
Comment (graiola) [inline paper.md]: the difference with pr2_controller_manager in the generality of the framework. 
ros-control is not binded to any specific robot.

Comment (graiola) [reply paper.md]: I would start by talking about the core idea of ros_control, and after compare it to pr2_controller_manager:
The idea of `ros_control` is to bring real time control capabilities into ROS (which does not have) while keeping the other cool ROS' features such as... . The idea originated from ... .

Comment (bmagyar) [reply paper.md]: check now

File: paper.md [hunk 2]
```diff
@@ -67,25 +67,27 @@ affiliations:
...
+The backbone of the framework is the Hardware Abstraction Layer, which serves as a bridge to different simulated and real robots. This abstraction is provided by the `hardware_interface::RobotHW` class; specific robot implementations have to inherit from this class. Instances of this class model hardware resources provided by the robot such as electric and hydraulic actuators and low-level sensors such as encoders and force/torque sensors. It also allows for mix and match heterogeneous hardware or swap out components transparently whether it is a real or simulated robot. Controllers are also hardware-agnostic. They specify required hardware resources to be able to operate. Their lifecycle is not static, and controller configurations can be queried/modified at runtime: start/stop, reconfigure.
```
Comment (graiola) [inline paper.md]: Thanks to this layer, controllers can be hardware-agnostic and easily sharable among different robots. 

I would remove "They specify required hardware resources to be able to operate." or rephrase it by talking about the hardware interfaces (could be useful to explain the image)

"Their lifecycle is not static, and controller configurations can be queried/modified at runtime: start/stop, reconfigure." should be moved in the controller_manager part

File: paper.md [hunk 3]
```diff
@@ -67,25 +67,27 @@ affiliations:
...
+There is a possibility for composing already implemented `RobotHW` instances which is ideal for constructing control systems for robots where parts come from different suppliers, each supplying their own specific `RobotHW` instance. The rest of the `hardware_interface` package defines read-only or read-write typed joint and actuator interfaces for abstracting hardware away. Through these typed interfaces this abstraction enables easy introspection and increases maintainability.
 
-The `controller_manager` is responsible for managing the lifecycle of controllers, and hardware resources through the interfaces and handling resource conflicts between controllers. It provides a standard `ROS service`-based interface for controller lifecycle management and queries.
+The `controller_manager` is responsible for managing the lifecycle of controllers, and hardware resources through the interfaces and handling resource conflicts between controllers. The lifecycle of controllers is not static. It can be queried and modified at runtime through standard `ROS services` provided by the `controller_manager`: start, stop, configure.
```
Comment (graiola) [inline paper.md]: start, stop, configure, etc...

File: paper.md [hunk 4]
```diff
@@ -67,25 +67,27 @@ affiliations:
...
 
-The backbone of the framework is the Hardware Abstraction Layer, which serves as a bridge to different simulated and real robots. This abstraction is provided by the `hardware_interface::RobotHW` class; specific robot implementations have to inherit from this class.  Instances of this class are then used to interface the robot hardware (including some low-level sensors) to higher-level controllers. This allows higher-level controllers to be hardware-agnostic and easily shareable.
+The backbone of the framework is the Hardware Abstraction Layer, which serves as a bridge to different simulated and real robots. This abstraction is provided by the `hardware_interface::RobotHW` class; specific robot implementations have to inherit from this class. Instances of this class model hardware resources provided by the robot such as electric and hydraulic actuators and low-level sensors such as encoders and force/torque sensors. It also allows for mix and match heterogeneous hardware or swap out components transparently whether it is a real or simulated robot. Controllers are also hardware-agnostic. They specify required hardware resources to be able to operate.
```
Comment (graiola) [inline paper.md:80]: since they are already in the images, we could refer to the interfaces, efforts, velocities, etc... ?

Comment (bmagyar) [reply paper.md:80]: Added a bit below

File: paper.md (hunk 3 above)
Comment (bmagyar) [reply paper.md]: done

**Pull request merged**